#27 yacht dice

AliceOfSNU opened the pull request:
client팀에서 만들어주신 미니게임인 yacht dice를 xtown월드에 집어넣고
월드에 놓여진 주사위 모양 포탈을 사용해 게임 씬을 불러오도록 작업했습니다.
멀티플레이 기능은 구현중이나 여러 이슈에 부딪혀서 3일까지 complete할지 미지수이나,
최대한 하는데까지 해보겠습니다.

지금은 singleplay로 플레이 가능합니다.
진홍님이 제가 코드 따 간 이후로 몇몇 change를 더하신 것 같은데 그부분 여기에도 반영해주시길 부탁드립니다.

AliceOfSNU committed:
yacht dice

AliceOfSNU committed:
adding pun and basic turn taking and messaging functionality

AliceOfSNU committed:
added multiplayer mode, but blocked it because it's still in development.
Currently only single play mode available

JasonK2K commented:
주사위 클릭 했을 때 닉네임 입력하는 UI 와 join UI 가 겹치는 것 같네요! 확인부탁드립니당

ahnjinmo commented:
> 주사위 클릭 했을 때 닉네임 입력하는 UI 와 join UI 가 겹치는 것 같네요! 확인부탁드립니당

이건 PlayGround scene의 Canvas 게임오브젝트의 Canvas Scaler가 아래 사진과 같이 Constant Pixel Size로 설정되어 있어서 플레이 화면의 해상도에 따라 UI 컴포넌트의 상대위치가 달라져서 그런 것 같습니다. Scale With Screen Size로 바꾸면 해결 될 것 같습니다.
![canvas](https://user-images.githubusercontent.com/89832796/147846981-a4dd4133-c567-4b5c-a0e6-71ef2fda7d35.png)

ahnjinmo committed:
Fixed error on small straight calculation

Fixed error on small straight calculation

ahnjinmo committed:
Fixed errors on score and dice selection

Blocked user from selecting score UI and dice when gamestate isn't in selecting state

ahnjinmo committed:
Cleaned up folders

Cleaned up folders

ahnjinmo commented:
#22 에 따라 디렉토리를 정리하였습니다. 하위폴더 이름은 모두 Yacht로 통일하였습니다.

AliceOfSNU commented:
아 넵 그럼 UI겹치는 문제는 지금 사진에 올려주신 것처럼 수정되서 해결 된 상태인가요?

ahnjinmo commented:
> 아 넵 그럼 UI겹치는 문제는 지금 사진에 올려주신 것처럼 수정되서 해결 된 상태인가요?

앗 아뇨 수정은 안 된 상태입니다! 제가 수정해서 추가로 commit 하겠습니다

ahnjinmo committed:
Fixed UI elements' overlapping issue

Fixed UI elements' overlapping issue

AliceOfSNU committed:
chat hide and show, prefabs edited

AliceOfSNU committed:
UI 수정

AliceOfSNU committed:
chat 서버 id 추가

AliceOfSNU committed:
removed debug.logs
